lib/heidrun/id_source.rb

```diff
@@ -0,0 +1,55 @@
+
+module Heidrun
+
+  ##
+  # A source of identifier strings to use in constructing API requests.
+  #
+  # This is a temporary trick to allow us to get around a limitation
+  # in NARA's API: we need to request a list of IDs explicitly because we can't
+  # page through their entire result set.  IDSource reads a file and gives us
+  # batches of IDs as arrays.
+  #
+  class IDSource
+
+    ##
+    # Constructor
+    #
+    # IDSource is initialized with the filehandle of a file that contains one
+    # ID string per line.
+    #
+    # The batchsize parameter will determine how many records are requested
+    # with a single request to the API.
+    #
+    # @param fh        File    File-like object (File / StringIO)
+    # @param batchsize Fixnum
+    #
+    def initialize(fh, batchsize = 10)
+      @fh = fh
+      @batchsize = batchsize
+    end
+
+    ##
+    # Return an enumerator of arrays of identifiers.
+    # @return Enumerator
+    def batches
+      en = Enumerator.new do |e|
+        batch = []
+        i = 1
+        @fh.each_line do |line|
+          batch << line.chomp.delete(',')
```

Comment:
See comment above. If this is a trailing `,` is expected for these files, we should either document it, or make this a `NARAIDSource` class.